packages/lib/src/interactions/DefaultInteractions.tsx

```diff
@@ -1,58 +1,80 @@
-import { merge } from 'lodash';
-
 import Pan from '../interactions/Pan';
+import type { PanProps } from '../interactions/Pan';
 import SelectToZoom from '../interactions/SelectToZoom';
+import type { SelectToZoomProps } from '../interactions/SelectToZoom';
 import XAxisZoom from '../interactions/XAxisZoom';
+import type { XAxisZoomProps } from '../interactions/XAxisZoom';
 import YAxisZoom from '../interactions/YAxisZoom';
+import type { YAxisZoomProps } from '../interactions/YAxisZoom';
 import Zoom from '../interactions/Zoom';
+import type { ZoomProps } from '../interactions/Zoom';
 import AxialSelectToZoom from './AxialSelectToZoom';
-import type { Interactions } from './models';
-import { getDefaultInteractions } from './utils';
+import type { AxialSelectToZoomProps } from './AxialSelectToZoom';
+
+export interface DefaultInteractionsConfig {
+  pan?: PanProps | false;
+  zoom?: ZoomProps | false;
+  xAxisZoom?: XAxisZoomProps | false;
+  yAxisZoom?: YAxisZoomProps | false;
+  selectToZoom?: Omit<SelectToZoomProps, 'keepRatio'> | false;
+  xSelectToZoom?: Omit<AxialSelectToZoomProps, 'axis'> | false;
+  ySelectToZoom?: Omit<AxialSelectToZoomProps, 'axis'> | false;
+  withAxialZoom?: boolean; // shortcut to turn off all axial zoom interactions
+}
 
-interface Props {
-  interactions?: Interactions;
+interface Props extends DefaultInteractionsConfig {
   keepRatio?: boolean;
 }
 
 function DefaultInteractions(props: Props) {
-  const { interactions: givenInteractions = {}, keepRatio } = props;
-
-  const parsedInteractions = Object.fromEntries(
-    Object.entries(givenInteractions).map(([k, v]) => {
-      if (v === true) {
-        return [k, {}];
-      }
-
-      if (v === false) {
-        return [k, null];
-      }
-
-      return [k, v];
-    })
-  );
-
-  const interactions = merge(
-    getDefaultInteractions(keepRatio),
-    parsedInteractions
-  );
```

Comment:
At first, I tried keeping this algorithm, but I was losing a lot of type strictness and had to do a lot of castings. I went with a more declarative version where each interaction is dealt with explicitly in the JSX.